### PR TITLE
Remove native camera scale matrix

### PR DESCRIPTION
**Pull request by GengineJS**

### CODE REVIEW

File: native/cocos/scene/Camera.cpp [hunk 1]
```diff
@@ -190,6 +190,9 @@ void Camera::update(bool forceUpdate /*false*/) {
 
     // view-projection
     if (viewProjDirty) {
+        Mat4 scaleMat{};
+        scaleMat.scale(_node->getWorldScale());
+        Mat4::multiply(scaleMat, _matView, &_matView);
```
Comment (minggo) [inline native/cocos/scene/Camera.cpp]: Have two questions:
- why need to remove scale matrix
- the logic seems to multiply scale matrix

Comment (GengineJS) [reply native/cocos/scene/Camera.cpp]: for this issue：https://github.com/cocos/3d-tasks/issues/10377

Comment (GengineJS) [reply native/cocos/scene/Camera.cpp]: Because _matView is the inverse world matrix of the node, it needs to be corrected by the world scale of the node

Comment (GengineJS) [reply native/cocos/scene/Camera.cpp]: as shown below：

image

Comment (minggo) [reply native/cocos/scene/Camera.cpp]: It will do reverse every frame. I think can cache the result. It is updated only `_matView` is changed.

Comment (GengineJS) [reply native/cocos/scene/Camera.cpp:195]: like this?

Mat4::multiply(scaleMat, _matView, &_tempMatView); ?

Comment (GengineJS) [reply native/cocos/scene/Camera.cpp]: The position has been changed, it will not update every frame now

File: native/cocos/scene/Camera.cpp [hunk 2]
```diff
@@ -160,7 +160,10 @@ void Camera::update(bool forceUpdate /*false*/) {
     if (_node->getChangedFlags() || forceUpdate) {
         _matView = _node->getWorldMatrix().getInversed();
         _forward.set(-_matView.m[2], -_matView.m[6], -_matView.m[10]);
-
+        Mat4 scaleMat{};
+        scaleMat.scale(_node->getWorldScale());
+        // remove scale
+        Mat4::multiply(scaleMat, _matView, &_matView);
```
Comment (minggo) [inline native/cocos/scene/Camera.cpp:166]: Why native implementation is different from ts version?

Comment (GengineJS) [reply native/cocos/scene/Camera.cpp:166]: native's dst is on the right, typescript is on the left. so it is consistent.

static void cc::Mat4::multiply(const cc::Mat4 &m1, const cc::Mat4 &m2, cc::Mat4 *dst)

**Review by minggo — APPROVED** [review]

**Review by star-e — APPROVED** [review]

**Pull request merged**